assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.test.js

```diff
@@ -16,47 +16,102 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import fetchMock from 'fetch-mock';
+
 /**
  * Internal dependencies
  */
 import { renderHook, actHook as act } from '../../../../../tests/js/test-utils';
-import { createTestRegistry } from '../../../../../tests/js/utils';
+import {
+	createTestRegistry,
+	waitForDefaultTimeouts,
+} from '../../../../../tests/js/utils';
 import { MODULES_TAGMANAGER } from '../datastore/constants';
 import { createBuildAndReceivers } from '../datastore/__factories__/utils';
 import useGAPropertyIDEffect from './useGAPropertyIDEffect';
 
 describe( 'useGAPropertyIDEffect', () => {
 	let registry;
-	beforeEach( () => {
-		registry = createTestRegistry();
-		// Set settings to prevent fetch in resolver.
-		registry.dispatch( MODULES_TAGMANAGER ).receiveGetSettings( {} );
-		// Set set no existing tag.
-		registry.dispatch( MODULES_TAGMANAGER ).receiveGetExistingTag( null );
-	} );
 
-	it( 'sets the gaPropertyID with the current detected singular property ID in selected containers', async () => {
-		const { buildAndReceiveWebAndAMP } =
-			createBuildAndReceivers( registry );
+	describe( 'with empty tagmanager settings store', () => {
+		beforeEach( () => {
+			registry = createTestRegistry();
+		} );
+
+		it( 'fetches settings from api before updating gaPropertyID', async () => {
+			fetchMock.getOnce(
+				/^\/google-site-kit\/v1\/modules\/tagmanager\/data\/settings/,
+				{
+					body: {},
+					status: 200,
+				}
+			);
+
+			await act(
+				() =>
+					new Promise( ( resolve ) => {
+						renderHook( () => useGAPropertyIDEffect(), {
+							registry,
+						} );
+						resolve();
+					} )
+			);
+
+			await act( waitForDefaultTimeouts );
 
-		const TEST_GA_PROPERTY_ID = 'UA-123456789-1';
+			expect( fetchMock ).toHaveFetched(
+				new RegExp(
+					'^/google-site-kit/v1/modules/tagmanager/data/settings'
+				)
+			);
 
-		buildAndReceiveWebAndAMP( {
-			webPropertyID: TEST_GA_PROPERTY_ID,
+			const propertyID = registry
+				.select( MODULES_TAGMANAGER )
+				.getGAPropertyID();
+
+			expect( propertyID ).toBe( '' );
```

Comment:
What exactly is this testing?

The test spec says it fetches from the API before setting/updating the value, but what was the value before? Why are we looking for an empty string for `propertyID`? I suspect it's because previously it was `undefined`, but because that isn't in this test anywhere it's hard-to-follow the logic here.

Could you expand on the test a bit, either with some comments highlighting what's being tested and happening "behind-the-scenes" of the selector calls, or with a few more `expect` statements?

Reply:
You are right, this does not make sense. So I removed this part, as setting the propertyID is already tested in the other test, so we don't need it here.

So this test is now only verifies the settings are loaded from the API if the settings are empty. Do you think this makes sense?

Reply:
That makes a lot more senes, yeah. Thanks so much 👍🏻